src/state/reducers.ts

```diff
@@ -1,10 +1,27 @@
-import {
-  combineReducers, Action, Reducer, ReducersMapObject,
-} from 'redux';
-import query from '../search/state/reducers';
-import results from '../results/state/reducers';
+import { combineReducers, Action, Reducer, ReducersMapObject } from 'redux';
+import query, { SearchAction } from '../search/state/reducers';
+import results, { ResultAction } from '../results/state/reducers';
+import { SearchState } from '../search/state/initialState';
+import { ResultsState } from '../results/state/initialState';
 
-export default combineReducers({
+type RootState =
+  | {
+      query: SearchState;
+      results: ResultsState;
+    }
+  | undefined;
+
+const appReducer = combineReducers({
   query,
   results,
 });
+
+const rootReducer = (state: RootState, action: SearchAction | ResultAction) => {
+  if (action.type === 'RESET') {
+    state = undefined;
```

Comment:
`null` is best used when we want to set a variable to no value

Reply:
The Dan Abramov explanation for doing this is with `undefined`, presumably because the variable is declared:
https://stackoverflow.com/questions/35622588/how-to-reset-the-state-of-a-redux-store/35641992#35641992

Reply:
Defining something as undefined seems counterintuitive to me but I'm sure it's all legit if Dan says it's ok.